### PR TITLE
docs for checkout billing agreement using native checkout

### PR DESCRIPTION
**Pull request by kgangineni**
### Summary of changes

- add docs for using Billing Agreement with native checkout

### Checklist

- ~[ ] Added a changelog entry~

@kgangineni 

-

### CODE REVIEW

**Review by jaxdesmarais — APPROVED** [review]
👻

File: docs/PayPalNativeCheckout/README.md [hunk 1]
```diff
@@ -155,6 +155,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
 --data-raw ''
 ```
 
+### Billing Agreement
```
Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md]: Add a link to this section in the begining of Readme

Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
## Billing Agreement
```

File: docs/PayPalNativeCheckout/README.md [hunk 2]
```diff
@@ -155,6 +155,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+
+## 1. Create Order
```
Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
### 1. Create Order
```

File: docs/PayPalNativeCheckout/README.md [hunk 3]
```diff
@@ -155,6 +155,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+```
+
+**Response**
+
+```json
+{
+   "token_id": "<TOKEN>"
+}
+```
+## 2. Set BillingAgreement token
```
Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
### 2. Set BillingAgreement token
```

File: docs/PayPalNativeCheckout/README.md [hunk 4]
```diff
@@ -155,6 +155,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+
+```swift
+payPalClient.start { createOrderAction in
+    createOrderAction.set(billingAgreementToken: "<billingAgreementToken>")
+}
+```
+## 3. Approve the order
```
Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md]:
```suggestion
### 3. Approve the order
```

**Review by mattwylder — APPROVED** [review]

File: docs/PayPalNativeCheckout/README.md [hunk 5]
```diff
@@ -5,8 +5,9 @@ The PayPal Native Checkout module in the PayPal SDK enables PayPal payments via
 Follow these steps to add PayPal Native Checkout payments:
 
 1. [Setup a PayPal Developer Account](#setup-a-paypal-developer-account)
-1. [Add PayPal Native Checkout Module](#add-paypal-native-checkout-module)
-1. [Test and go live](#test-and-go-live)
+2. [Add PayPal Native Checkout Module](#add-paypal-native-checkout-module)
+3. [Test and go live](#test-and-go-live)
+4. After initial setup, Follow instructions [here](#billing-agreement) for Billing Agreements
```
Comment (jcnoriega) [inline docs/PayPalNativeCheckout/README.md:10]:
```suggestion
4. After initial setup, follow instructions [here](#billing-agreement) for Billing Agreements
```

**Review by jcnoriega — APPROVED** [review]

File: docs/PayPalNativeCheckout/README.md [hunk 6]
```diff
@@ -155,6 +156,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+  {
+    "payment_method": "PAYPAL"
+  },
+  "plan":
+  {
+    "type": "MERCHANT_INITIATED_BILLING",
+    "merchant_preferences":
+    {
+      "return_url": "https://example.com/return",
+      "cancel_url": "https://example.com/cancel",
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md:188]: I don't think we need merchants to pass these in their create order request anymore.

File: docs/PayPalNativeCheckout/README.md [hunk 7]
```diff
@@ -155,6 +156,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+    }
+  }
+}'
+```
+
+**Response**
+
+```json
+{
+   "token_id": "<TOKEN>"
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md:202]: Nitpick - use the same variable here as below
```suggestion
   "token_id": "<BILLING_AGREEMENT_TOKEN>"
```

File: docs/PayPalNativeCheckout/README.md [hunk 8]
```diff
@@ -155,6 +156,62 @@ curl --location --request POST 'https://api.sandbox.paypal.com/v2/checkout/order
...
+}
+```
+### 2. Set BillingAgreement token
+
+```swift
+payPalClient.start { createOrderAction in
+    createOrderAction.set(billingAgreementToken: "<billingAgreementToken>")
```
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md:209]: Nitpick - use the same variable
```suggestion
    createOrderAction.set(billingAgreementToken: "<BILLING_AGREEMENT_TOKEN>")
```

File: docs/PayPalNativeCheckout/README.md (hunk 5 above)
Comment (scannillo) [inline docs/PayPalNativeCheckout/README.md:10]: Billing agreement here should come before testing and go live, to match the order of the doc below

**Pull request merged**